applications/CompressiblePotentialFlowApplication/tests/potential_flow_test_factory.py

```diff
@@ -79,9 +95,28 @@ def _run_test(self):
         potential_flow_analysis = PotentialFlowAnalysis(model,ProjectParameters)
         potential_flow_analysis.Run()
 
-class Naca0012SmallTest(PotentialFlowTestFactory):
-    file_name = "naca0012_small"
-    work_folder = "naca0012_small_test"
+@UnitTest.skipUnless(has_hdf5_application,"Missing required application: HDF5Application")
+@UnitTest.skipUnless(has_structural_application,"Missing required application: StructuralMechanicsApplication")
+class Naca0012SmallAdjointTest(PotentialFlowTestFactory):
```

Comment:
if instead of making a new class you add the test to the same class then they will be detected automatically

Reply:
Can you elaborate a bit more? Are you referring to the @UnitTest.skipUnless calls?

Reply:
ok here it comes (brief version, you can also read the docs of python-unittest):

if you have a class deriving from Kratos-Unittest, then all functions that start with `test...` will be recognized by the testrunner automatically

i.e. for your case I recommend the following:
~~~py
class Naca0012SmallTests(KratosUnittest):
    def NormalTest()
        ...

    def AdjointTest()
        If missing dependencies => self.skipTest()
        ...
~~~

and in `test_PotentialApp.py`:
~~~py
smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([Naca0012SmallTests]))# this automatically adds all tests
~~~

I suggest you take a look at the tests in the test_StructuralMechanicsApplication.py, where we use this a lot

I think you will need to refactor the current test a bit but I think this should be ok for @inigolcanalejo 

Also take a look at [this](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MeshMovingApplication/tests/test_ale_fluid_solver.py) test, this is really up-to-date and you can use it as a reference